filebeat/publish/publish.go

```diff
@@ -172,19 +172,28 @@ func (p *asyncLogPublisher) Start() {
 			case <-p.done:
 				return
 			case events := <-p.in:
-				pubEvents := make([]common.MapStr, len(events))
-				for i, event := range events {
+
+				tmpEvents := make([]*input.Event, 0)
+				for _, event := range events {
+					if event.HasData() {
+						tmpEvents = append(tmpEvents, event)
+					}
+				}
```

Comment:
we need to copy here? the events array is store in `eventsBatch` for pushing acked events(batches) to registrar. Interestingly the syncPublisher does forward all events without filtering.